### DISCUSSION TITLE
On Demand ISR

### DISCUSSION THREAD

**Discussion by iankduffy** [opennextjs/opennextjs-netlify#1228] [General]
Hello, NextJS 12.1 has added [On-demand Incremental Static Regeneration](https://nextjs.org/blog/next-12-1#on-demand-incremental-static-regeneration-beta) currently in Beta / Unstable, just wondering if in the future if we are looking to add this to netlify plugin?

**Comment by kamerat** [comment]
It seems they are going to support it, but they do not have a timeframe for it yet: https://github.com/netlify/netlify-plugin-nextjs/issues/1288#issuecomment-1085701256

**Reply by ascorbic** [reply]
That is correct.

**Reply by konstantinbrazhnik** [reply]
Boy oh boy would this feature really close the loop on NextJS on Netlify being the best static, headless configuration component on the web. Really hoping it's high up on the priority list.

**Reply by yescine** [reply]
this will unlock the ultimate productivity with webhooks coming from all over CMS requesting to change the content of NextJs app. what a time to work in web development !

**Reply by nickytonline** [reply]
It's currently in progress! https://github.com/netlify/netlify-plugin-nextjs/issues/1288#issuecomment-1193739702

**Reply by StefKors** [reply]
@nickytonline we just found out Netlify doesn't support on-demand ISR, how close are we to Netlify supporting this?

**Reply by nickytonline** [reply]
I don't have a timeline for this, but just keep an eye on this discussion.

**Reply by charkour** [reply]
Hey @nickytonline, is there any public timeline for the netlify nextjs runtime (not specific to On Demand ISR)? Thanks.

**Reply by orinokai** [reply]
Hey @charkour, sorry we don't currently keep a public timeline for the Next.js Runtime. We are currently awaiting some new platform features before we can implement on-demand ISR, but we're very keen to get it shipped asap.

**Reply by charkour** [reply]
Hi @orinokai, thanks. 

To get a rough idea of what the status is for this feature, what does "shipped asap" mean to you? Will this be a Q1-Q2 feature or a 2024 feature? I ask because every team has a slightly different definition of what "asap" means to them. Thank you.

**Reply by orinokai** [reply]
Hi @charkour. Work on this feature is currently in progress, so I hope that rules out a 2024 release 😅 Unfortunately I can't share any more specific details at this stage, but it's coming soon.

**Reply by charkour** [reply]
Excellent, any information is helpful. I appreciate that. Thank you

**Reply by Designer023** [reply]
soon™️ 🤣 

Thanks for the update. This is the biggest feature I am looking forward to this year 😁
